src/execution_result.rs

```diff
@@ -59,22 +59,28 @@ impl ContractExecutionResult {
                             } else {
                                 Err(Error::UnexpectedValue(format!(
                                     "wrong type, expect: array, value: {:?}",
-                                    value
+                                    &fields[0]
```

Comment:
I can't seem to understand why was this changed from `value` to `&fields[0]`.

Reply:
This error is thrown when value doesn't match `Struct { Struct { Array[felt252's] } }`. My impression here is that in case of failure, we weren't correctly 
indicating the issue. Example if `value = Struct { Struct { Uint8() } }`, the thrown error used to be `wrong type, expect: array, value: Struct { Struct { Uint8() } }`.
With the update to the error, it would look like `wrong type, expect: array, value: Uint8()`. I feel like this is a better indication. Let me know if you want me
to revert though.

Reply:
Ok I think I understand now, but I still find it confusing unless you already know what the error means (the subtle differences in the error message).

What about keeping the original value (with the complete field) so that we know it's always complete, but adding more context to the error message itself? Something like the following:

```rs
format!("wrong type, expected: Struct { Struct { Array<felt252> } }, value: {value:?}")
```

This way it has the complete value and the complete expected type so that the user can understand exactly where it failed. What do you think?

Reply:
this makes more sense, will update!